packages/core/src/external/carequality/ihe-gateway-v2/outbound/xcpd/process/xcpd-response.ts

```diff
@@ -304,25 +224,37 @@ export function processXCPDResponse({
   });
 
   const jsonObj = parser.parse(response);
-  const { ack, queryResponseCode } = getAckAndQueryResponseCodeFromPatientRegistryProfile(jsonObj);
-
-  if (isApplicationAccept(ack) && isXCPDRespOk(queryResponseCode)) {
-    log(`Found a match for cxId: ${cxId} patient: ${patientId}`);
-    return handlePatientMatchResponse({
-      jsonObj,
-      outboundRequest,
-      gateway,
-    });
-  } else if (isApplicationAccept(ack) && isXCPDRespNotFound(queryResponseCode)) {
-    return handlePatientNoMatchResponse({
-      outboundRequest,
-      gateway,
-    });
-  } else {
-    return handlePatientErrorResponse({
-      jsonObj,
+  try {
+    const iti55Response = iti55Schema.parse(jsonObj);
+    const patientRegistryProfile = getPatientRegistryProfile(iti55Response);
+    const { ack, queryResponseCode } =
+      getAckAndQueryResponseCodeFromPatientRegistryProfile(patientRegistryProfile);
+
+    if (isApplicationAccept(ack) && isXCPDRespOk(queryResponseCode)) {
+      log(`Found a match for cxId: ${cxId} patient: ${patientId}`);
+      return handlePatientMatchResponse({
+        patientRegistryProfile,
+        outboundRequest,
+        gateway,
+      });
+    } else if (isApplicationAccept(ack) && isXCPDRespNotFound(queryResponseCode)) {
+      return handlePatientNoMatchResponse({
+        outboundRequest,
+        gateway,
+      });
+    } else {
+      return handlePatientErrorResponse({
+        patientRegistryProfile,
+        outboundRequest,
+        gateway,
+      });
+    }
+  } catch (error) {
+    log(`Error processing XCPD response: ${error}`);
```

Comment:
nit: if we can, it would be good to use `errorToString()` that we can keep improving to log useful information (e.g., on [this PR](https://github.com/metriport/metriport/pull/2272/files#diff-435d3a78fb1ddbdf5a45825dc26b133334a10acd3e2c838959d78f4a16c2efb9R16) we include Axios' error `code` if available, that sometimes Axios itself doesn't include).